controller/kubernetes_pv_controller.go

```diff
@@ -512,31 +512,34 @@ func (kc *KubernetesPVController) cleanupVolumeAttachment(pods []*v1.Pod, volume
 	switch deletionStrategy {
 	case types.VolumeAttachmentDeletionNever:
 		// kubernetes default is to never remove a volume attachment from a downed node
-		waitForDeletion = true
+		waitForDeletion = len(terminatingPods) > 0
 	case types.VolumeAttachmentDeletionWait:
 		// in the longhorn default mode for safety reasons we wait till the deletion has passed
 		// this should lead to a force delete by the kubelet, but since the pod is still available
 		// we know that the kubelet failed to cleanup the pod resource.
 		for _, p := range terminatingPods {
-			waitForDeletion = waitForDeletion || !p.DeletionTimestamp.After(time.Now())
+			waitForDeletion = waitForDeletion || p.DeletionTimestamp.After(time.Now())
```

Comment:
?

Reply:
Read like this: 
If there are any pods that are still within the deletion grace period we need to wait for the pods deletion (`waitForDeletion`). Before trying to delete the VA.

Reply:
I will rename to `waitingForPodDeletion`